packages/catppuccin-vsc/src/theme/tokens/haskell.ts

```diff
@@ -0,0 +1,85 @@
+import type { TextmateColors, ThemeContext } from "@/types";
+
+const tokens = (context: ThemeContext): TextmateColors => {
+  const { palette } = context;
+
+  /*
+  should preferrably appear visually distinct:
+  - data constructors -> use blue
+  - type constructors -> use yellow+italic
+  - typeclasses       -> use yellow
+  */
+
+  return [
+    {
+      name: "data constructors",
+      scope: ["meta.declaration.data constant.other", "constant.other"],
+      settings: { foreground: palette.blue },
+      // like functions (capitalized -> still visually distinct)
+    },
+    {
+      name: "type constructors",
+      scope: ["storage.type.haskell"],
+      settings: {
+        foreground: palette.yellow,
+        fontStyle: "italic",
+      },
+    },
+    {
+      name: "typeclasses",
+      scope: ["entity.name.type.class"],
+      settings: {
+        foreground: palette.yellow,
+        fontStyle: "",
+      },
+    },
```

Comment:
Everything in this file should be specific to haskell in the textmate scopes. This generic scope will affect other languages outside of haskell and should be carefully introduced in the `index.ts` file if need be.

I'd like to see this removed in favour of a more specific scope.